File: LicentaInAngular.Server/DataLayer/Models/Cos.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace LicentaInAngular.Server.Models
{
    public class Cos
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int idCos { get; set; }

        [Required]
        public string CodUnic { get; set; }

        [ForeignKey("User")]  // ForeignKey to User (Admin)
        public int IdUser { get; set; }  // Nullable ForeignKey to User (Admin)

        public User? User { get; set; }

        [Required]
        public DateTime DataCreare { get; set; } // To track creation time for automatic deletion
    }
}

File: licentainangular.client/src/app/models/Cos.ts
import { User } from './User';
export interface Cos {
  idCos: number; //Sau idCos
  CodUnic: string;
  IdUser: number;
  User: User | null;  // Ideally, replace 'any' with the actual User model
  DataCreare: Date;
}
